include/maidsafe/crux/socket.hpp

```diff
@@ -60,8 +60,7 @@ class socket
         typename boost::asio::handler_type<CompletionToken,
                                            void(boost::system::error_code)>::type
         >::type
-    async_connect(const endpoint_type& remote_endpoint,
-                  CompletionToken&& token);
+    async_connect(endpoint_type remote_endpoint, CompletionToken&& token);
```

Comment:
I only changed it because I wanted to modify its address, I could have modified the `remote` variable directly, not sure why I didn't, I guess at the back of my head I thought maybe the original value that was passed as an argument should be stored in the `remote` variable, but at the end, it is not the case in the commited code anyway.

Other than that, I'm all right with `endpoint_type` being passed as const ref.